### DISCUSSION TITLE
Issue with Leaflet Map on Blues Stack

### DISCUSSION THREAD

**Discussion by jotthyr** [remix-run/remix#6034] [Q&A] [answered]
**Have you experienced this bug with the latest version of the template?**

Yes, Blues Stack

**Steps to Reproduce**

- Install leaflet, react-leaflet as dependency (try both 4.x.x or 3.x.x)

- Implement code from remix.run example:

https://codesandbox.io/s/github/remix-run/examples/tree/main/leaflet

- There are two kind of errors:

  A. error implementing react-leaflet 4.x.x says that cannot use ESModules from this package

  A. then error says that, window is not defined
  
  B. error implementing react-leaflet 3.x.x says only that, window is not defined

**Expected Behavior**

Load map only on client side to be able to relate to window

-----------------------------------------------------------------------
**Questions:**

1. Is there any way to spin up this leaflet map? I have tried many workarounds and didn't worked
2. In next.js there is utility that help solving this issue, is any in remix? 
```
import dynamic from 'next/dynamic'

const DynamicComponentWithNoSSR = dynamic(() => import('../components/List'), {
  ssr: false
})

export default () => <DynamicComponentWithNoSSR />
```
Any ideas?

**Comment by gongmoga237** [comment, marked as answer]
Hello, i have encoutered this issue when trying to integrate tom tom map in my remix project, the trick i used to fix that was to name my component **Component.client.tsx** and then after importing it in my route component i used ClientOnly  from "remix-utils" npm package 
to wrap my **Component** that's will render the component when hydration have been done. It just worked fine after that.

**Reply by jotthyr** [reply]
Thanks! small detail to add .client.tsx
